### PR TITLE
Feature/gh 1517 dynamic ui rendering

### PR DESCRIPTION
**Pull request by wlycdgr**
Implements dynamic updating of the extension panel UI including the Summary and Detailed views. This includes dynamic updating of the various tracker counts as the page loads, dynamic insertion of trackers into the blocking list as they are found, and dynamic redrawing of the donut graph.

Streamlines all UI data transfer between the panel (including its Panel, Summary, Blocking, Settings, and Rewards components) and the background to use ports managed by PanelData, obviating all such transfer through message passing.

Factors out Cliqz antitracking and adblocking modules data retrieval into a utility file that both BrowserButton and PanelData can use, reducing duplication across BrowserButton and PanelData & the GET_CLIQZ_MODULE_DATA reducer on the front end. 

Refactors FoundBugs to more efficiently handle the now potentially much larger volume of requests for bug info, primarily by storing foundApps data alongside foundBugs data so it does not need to be repeatedly recalculated.

Refactors and simplifies PanelData.

* [X] Have you followed the guidelines in [CONTRIBUTING.md](../CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you added an explanation of what your changes do?
* [X] Does your submission pass tests?
* [X] Did you lint your code prior to submission?

### CODE REVIEW

File: app/panel/components/Blocking.jsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ class Blocking extends React.Component {
 		this.state = {
 			blockingClasses: '',
 			disableBlocking: false,
+			isCategoryExpanded: new Map()
```
Comment (IAmThePan) [inline app/panel/components/Blocking.jsx]: I'm still not seeing that the Categories stay expanded/collapsed

File: app/panel/components/BuildingBlocks/DonutGraph.jsx [hunk 2]
```diff
@@ -159,37 +223,31 @@ class DonutGraph extends React.Component {
...
-			.enter().append('g')
-			.attr('class', 'arc');
+		// CONNECT NEW DATA
+		const arcs = this.chartCenter.selectAll('g')
+			.data(this.trackerPie(graphData), d => d.data.id);
+
+		// UPDATE
+		arcs.select('path')
+			.transition()
+			.duration(d => (d.value / totalTrackers) * animationDuration)
```
Comment (IAmThePan) [inline app/panel/components/BuildingBlocks/DonutGraph.jsx]: I'm seeing the graph update right away.  I'm not seeing an animation transition.  Is this expected?

File: app/panel/components/BuildingBlocks/DonutGraph.jsx [hunk 3]
```diff
@@ -159,37 +223,31 @@ class DonutGraph extends React.Component {
...
+		.append('path')
 			.style('fill', (d, i) => {
 				if (renderGreyscale) {
-					const greyTone = graphData.length > 1 ? 100 / (graphData.length - 1) * i * 0.01 : 0;
-					return this.colors.greyscale(greyTone);
+					return this.colors.greyscale(this.getTone(categoryCount, i));
 				} else if (renderRedscale) {
-					const redTone = graphData.length > 1 ? 100 / (graphData.length - 1) * i * 0.01 : 0;
-					return this.colors.redscale(redTone);
+					return this.colors.redscale(this.getTone(categoryCount, i));
```
Comment (IAmThePan) [inline app/panel/components/BuildingBlocks/DonutGraph.jsx]: The colors don't change when the panel is open and I pause or restrict Ghostery.  But they do if I close the panel and reopen it.

File: app/panel/components/BuildingBlocks/DonutGraph.jsx (hunk 2 above)
Comment (IAmThePan) [reply app/panel/components/BuildingBlocks/DonutGraph.jsx]: The colors do change if I switch from simple to detailed view.

**Comment by wlycdgr** [comment]
#1 (tests failing) and #5 (category list expand/collapse persistence on update) have been addressed

**Comment by wlycdgr** [comment]
3 (redscale/greyscale realtime updating) has been addressed

**Comment by wlycdgr** [comment]
1 through 5 have all been addressed

**Comment by jsignanini** [comment]
Closing in favor of #355

**Pull request closed without merging**